### ISSUE TITLE
Ideas for the integration of BrainGlobe atlases within napari 

### ISSUE THREAD

**Issue by adamltyson** [brainglobe/brainglobe-napari#1]
Currently BrainGlobe atlases are used by napari plugins (e.g. `brainreg`, `brainreg-segment`), but these are not accessible by napari plugins directly (only by using `bg-atlasapi` in the plugin code). It would be useful to be able to load, visualise and manipulate these atlases for many reasons, e.g.:
* To visualise atlas data
* To use the atlases as a target for third party registration libraries
* To define coordinate spaces for other analysis tools

As a first pass, my idea is to create a napari plugin that will include some of the functionality of the brainglobe atlas API and CLI, including:
* Query available atlases
* Inspect atlas metadata
* Load (and visualise) atlases

`bg-atlasapi` represents atlases as an abstract object. It would also be useful to do the same within napari. i.e. represent all the information within an atlas object as a "thing". It won't be a new layer type, but as close to this as possible (layer group?).

A sketch of my idea:
![brainglobe-napari-sketch](https://user-images.githubusercontent.com/13147259/221590567-f5036270-5346-4e58-9455-eca8a9808896.png)

cc @alessandrofelder @dstansby

**Comment by alessandrofelder** [comment]
What should this napari plugin be called? `BrainGlobe-Atlas-Viewer`?

**Comment by adamltyson** [comment]
I like `BrainGlobe-Atlas-Viewer`. It could live in this repo though, and `pip install brainglobe-napari` would install a number of related plugins.

**Comment by alessandrofelder** [comment]
The[ Layer group PR to Napari ](https://github.com/napari/napari/pull/5132) has not been merged yet.
LayerList cannot fulfil the functionality we need (it's not a Layer, and therefore we can't have nested LayerLists).

A possible temporary approach would be to encapsulate the conversion from BG atlas api Atlases to our napari atlas type in just one place, so we can easily change how this is done later? For now, this would just add appropriate layers for each of the images and meshes to napari?

**Comment by adamltyson** [comment]
> A possible temporary approach would be to encapsulate the conversion from BG atlas api Atlases to our napari atlas type in just one place, so we can easily change how this is done later? For now, this would just add appropriate layers for each of the images and meshes to napari?

Yeah I think so, unless there's something we can do to help push the layer group PR forwards.

**Comment by alessandrofelder** [comment]
@adamltyson what are your thoughts on when/how users can add meshes as napari surface layers?
Naive possibilities are:
1. add all meshes as layers by default (but that can be slow and lots of layers can be unwieldy in napari (although having nested layers would help a bit)! Initial experiments suggest even slower if we want to colour the meshes by their level in the hierarchy)
2. only add the root mesh, and have some widgets (a drop down? maybe a search field?) that allow you to add specific meshes you care about?
3. have a slider with the level of granularity you want to add meshes for. I.e. if the slider is on 3, you add all meshes that are grandchildren of the root mesh (either via the add to viewer button or a separate add meshes button)?

**Comment by adamltyson** [comment]
I'm leaning towards 2. It could be a tree like structure with checkboxes. I think this is what the brainrender GUI uses.

**Comment by adamltyson** [comment]
I'm also not sure 3. would be much use to people. I can see instances when it might be helpful to visualise a sub-sub division of area A along with the whole of area B.

**Comment by adamltyson** [comment]
@alessandrofelder this can be closed now right?
